test/cql-pytest/test_warnings.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Tests if warnings are returned in client's response
+#############################################################################
+
+from util import unique_name
+
+import pytest
+
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (k int primary key, t text)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+
+@pytest.fixture(scope="module")
+def generate_batch_func(table1):
+    """generates function for batch string generation.
+    Size in kb is valid when Scylla uses default settings."""
+
+    def func(size_in_kb):
+        batch_string = "BEGIN BATCH\n"
+        statements = []
+        for idx in range(size_in_kb):
+            statements.append(f"INSERT INTO {table1} (k, t) VALUES ({idx}, '{'x'*743}')")
+        batch_string += "\n".join(statements) + "\n APPLY BATCH\n"
+        return batch_string
+    return func
+
+
+def test_warnings_are_returned_in_response(cql, table1, generate_batch_func):
+    """Verifies if response contains warning messages.
+    Because warning message threshold is different for Scylla and Cassandra,
+    test tries several sizes until warning appears.
+    """
+    for size_in_kb in [10, 129, 256]:
+        oversized_batch = generate_batch_func(size_in_kb=size_in_kb)
+        response_future = cql.execute_async(oversized_batch)
+        response_future.result()
+        if response_future.warnings:
+            break
```

Comment:
If none of these created a warning, i.e., no "break" was called, your code will fall through and if I understand correctly, will fails weirdly on warnings[0] not existing. Maybe it's better to more explicitly add an "else" to the for, which is called if none of the break calls happened. E.g.,

```python
for size_in_kb in ...:
    ...
else:
   pytest.fail('Oversized batch did not generate a warning')
```

Reply:
Good idea. Thanks!